webknossos/webknossos/dataset/dataset.py

```diff
@@ -954,6 +955,18 @@ def add_layer_like(self, other_layer: Layer, layer_name: str) -> Layer:
         self._export_as_json()
         return self._layers[layer_name]
 
+    def _add_existing_layer(self, layer_properties: LayerProperties) -> Layer:
```

Comment:
I feel like something like this already exists somewhere :thinking: but didn’t find it just now, so maybe I dreamt that (or it was removed in the meantime)

Reply:
There is a method called `add_layer_for_existing_files`. However, it works differently because it tries to guess all the parameters from the data. The new method expects all parameters to be determined already.